src/plugins/lua/bindings/lua_ImGuiCustom.cpp

```diff
@@ -99,6 +102,39 @@ void RegisterBindings_ImGuiCustom(sol::table& ImGui)
 			[](mq::imgui::ConsoleWidget* pThis, const ImVec4& col, std::string_view text) { pThis->AppendText(text, MQColor(col), false); }
 		)
 	);
-}
 
+	// Widgets: ZepInputBox
+	ImGui.new_usertype<mq::imgui::ImGuiZepEditor>(
+		"TextEditor", sol::factories(&mq::imgui::ImGuiZepEditor::Create),
+		"Render", sol::overload(
+			[](mq::imgui::ImGuiZepEditor* pThis) { pThis->Render(); },
+			[](mq::imgui::ImGuiZepEditor* pThis, ImVec2 displaySize) { pThis->Render(displaySize); }
+		),
+		"Clear", &mq::imgui::ImGuiZepEditor::Clear,
+		"SetSyntax", [](mq::imgui::ImGuiZepEditor* pThis, const char* syntaxName) { pThis->SetSyntaxProvider(syntaxName); },
+		"text", sol::property(
+			[](mq::imgui::ImGuiZepEditor* pThis) -> std::string { size_t bufferSize = pThis->GetTextLength(); std::string text; text.reserve(bufferSize); pThis->GetText(text); return text; }
+			, &mq::imgui::ImGuiZepEditor::SetText),
```

Comment:
I didn't want to make this a property because it would be easy to mis-use and set the text without realizing its side effects.

If we really want this to be a property, then make it read only, and add a "LoadContents" or similarly named function that indicates we're not just setting a string.

Reply:
Fixed.